src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/service/NonAssociationMappingService.kt

```diff
@@ -153,4 +162,28 @@ class NonAssociationMappingService(
     } ?: run {
       nonAssociationMappingRepository.deleteAll()
     }
+
+  @Transactional
+  suspend fun updateMappingsByNomisId(oldOffenderNo: String, newOffenderNo: String) {
+    nonAssociationMappingRepository.findByFirstOffenderNoOrSecondOffenderNo(oldOffenderNo, oldOffenderNo)
+      .forEach {
+        log.info("Merging from $oldOffenderNo to $newOffenderNo: processing mapping $it")
+
+        if (oldOffenderNo == it.firstOffenderNo) {
+          if (newOffenderNo == it.secondOffenderNo) {
+            throw ValidationException("Found NA clash in $it when updating offender id from $oldOffenderNo to $newOffenderNo")
+          }
+          // it.firstOffenderNo = newOffenderNo
+          nonAssociationMappingRepository.updateFirstOffenderNo(it.nonAssociationId, newOffenderNo)
+
+        } else if (oldOffenderNo == it.secondOffenderNo) {
+          if (it.firstOffenderNo == newOffenderNo) {
+            throw ValidationException("Found NA clash in $it when updating offender id from $oldOffenderNo to $newOffenderNo")
+          }
+          // it.secondOffenderNo = newOffenderNo
+          nonAssociationMappingRepository.updateSecondOffenderNo(it.nonAssociationId, newOffenderNo)
+        }
+        // nonAssociationMappingRepository.save(it)
+      }
```

Comment:
I considered doing all this with 2 big updates but this method means more detailed logging